### DISCUSSION TITLE
Windows 11 has removed the "Legacy Console" option

### DISCUSSION THREAD

**Discussion by pcmicro** [leecher1337/ntvdmx64#250] [General]
The "Legacy Console" option was removed from Windows 11 in December 2023. Is there a way to recover it once the Windows update has removed it?

**Comment by leecher1337** [comment]
That suprises me, because I thought they made it an optional component? At least they said so:
https://github.com/microsoft/terminal/issues/15038#issuecomment-1482004895

Can you check if it is listed in optional components, maybe?

If not, I guess Win 11 is 23H2 dead end then, they also removed WOW16 support in USER32, so Windows then reached a point where it is too broken for NTVDM. The console could thoretically be extended to support the legacy console functions, as it is opensource on https://github.com/microsoft/terminal/
However, it's a bunch of overly-complex, unreadable horrific C++ code, so that I'm not able to even understand it, as it's also not very well documented. 

So the only hope is that someone is willing to implement ConhostV1 features into ConhostV2 or try to get the old ConhostV1 DLL running in 23H2 (that's what I could try in case it's not available as optional component).

Any volunteers for ConhostV2 expansion? Would also be benificial for users of WSL that relies on ConhostV2.

**Comment by leecher1337** [comment]
I tried to update win 11 VM to the latest release and I can't see a removal of Legacy Console so far, it's still there and working fine for me:

![image](https://github.com/leecher1337/ntvdmx64/assets/6495842/e031c518-2caf-4ac0-a1c3-5050b9ceb3b7)

**Comment by Peter0x44** [comment]
I put a post detailing how to reenable the legacy console on my blog.

https://peter0x44.github.io/posts/windows11-legacy-console/
